src/assoc.js

```diff
@@ -22,10 +21,14 @@ var _curry3 = require('./internal/_curry3');
  *      R.assoc('c', 3, {a: 1, b: 2}); //=> {a: 1, b: 2, c: 3}
  */
 module.exports = _curry3(function assoc(prop, val, obj) {
-  var result = {};
-  for (var p in obj) {
-    result[p] = obj[p];
+  var result = Object.create(Object.getPrototypeOf(obj))
+  var properties = Object.getOwnPropertyNames(obj)
```

Comment:
`Object.getOwnPropertyNames` gets non-enumerable props. If you go down that route you'll want to use `Object.defineProperty` and `Object.getOwnPropertyDescriptor` to ensure consistency.